src/FavIconCache.vala

```diff
@@ -47,13 +47,43 @@ public class FeedReader.FavIconCache : GLib.Object {
 			}
 			var enumerator = iconDirectory.enumerate_children(GLib.FileAttribute.STANDARD_NAME, 0);
 			GLib.FileInfo? fileInfo = null;
+			GLib.File? file = null;
 
-			while((fileInfo = enumerator.next_file()) != null)
+			while(true)
 			{
-				string fileName = fileInfo.get_name();
-				var pixbuf = new Gdk.Pixbuf.from_file_at_scale(iconDirPath + fileName, 24, 24, true);
-				fileName = fileName.substring(0, fileName.length - ".ico".length);
-				m_map.set(fileName, pixbuf);
+				if(enumerator.iterate(out fileInfo, out file))
+				{
+					if(fileInfo == null)
+						break;
+
+					string fileName = fileInfo.get_name();
+
+					if(fileName.has_suffix(".txt"))
+						continue;
+
+					try
+					{
+						var pixbuf = new Gdk.Pixbuf.from_file(iconDirPath + fileName);
+
+						if(pixbuf.get_height() <= 1 && pixbuf.get_width() <= 1)
+						{
+							Logger.warning(@"$fileName too small");
+							continue;
+						}
+
+						pixbuf = new Gdk.Pixbuf.from_file_at_scale(iconDirPath + fileName, 24, 24, true);
+						fileName = fileName.substring(0, fileName.length - ".ico".length);
+						m_map.set(fileName, pixbuf);
+					}
+					catch(GLib.Error e)
+					{
+						Logger.warning("Error loading favicon " + fileInfo.get_name());
+					}
+				}
+				else
+				{
+					Logger.warning("Error loading favicon " + fileInfo.get_name());
+				}
```

Comment:
No this is just a warning. I had some corrupt files (got them from tt-rss) and I just wanted to know what's going on.
I tried both ways of iterating through the files (had a bug and thought it was related to this code). I personally like this way better.


> With next_file, the gboolean return value signifies "end of iteration or error", which requires allocation of a temporary Error.
In contrast, with this function, a false return from iterate *always* means "error". End of iteration is signaled by out_info or out_child being null.